### ISSUE TITLE
Clean up deprecated Python TF code

### ISSUE THREAD

**Issue by tommadams** [tensorflow/minigo#740]
TF 1.13.0 marked a lot of stuff as deprecated. I've spotted these so far:

- tf.python_io.tf_record_iterator (replace with eager execution & TFRecordDataset)
- tensorflow.python.framework.ops.colocate_with (colocations handled automatically by placer)
- tensorflow.python.layers.normalization.batch_normalization (replace with keras.layers.batch_normalization)
- tensorflow.python.layers.core.dense (replace with keras.layers.dense)
- tensorflow.python.ops.math_ops.to_float (replace with tf.cast)
- tensorflow.python.ops.math_ops.to_int64 (replace with tf.cast)

There may be more

**Comment by brilee** [comment]
I plan to go through and rewrite all the TF stuff for 2.0 at some point, so don't worry too much about the deprecation stuff. The TPU changes from Estimator -> DistributionStrategy and associated hackery with the .pb exporting will overwhelm any other changes you were thinking of.

**Comment by l1t1** [comment]
keras is ugly

**Comment by l1t1** [comment]
https://github.com/tensorflow/docs/blob/master/site/en/r2/guide/upgrade.md
